lib/storage/src/content_manager/consensus_manager.rs

```diff
@@ -789,7 +793,8 @@ impl<C: CollectionContainer> ConsensusManager<C> {
 
         let first_unapplied_index = applied_index + 1;
 
-        debug_assert!(first_unapplied_index <= first_entry.index);
+        // ToDo: it seems like a mistake, need to check if it's correct
+        // debug_assert!(first_unapplied_index <= first_entry.index);
```

Comment:
This part need extra review

Reply:
I agree. I think this is only valid if we always compact all entries.

Doesn't it need to be like this?

```suggestion
        // debug_assert!(first_unapplied_index >= first_entry.index);
```